drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_auxpll
+
+#include <stdint.h>
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/toolchain.h>
+
+#include <hal/nrf_auxpll.h>
+
+/* obtain range value from configuration */
+#define AUXPLL_RANGE_FROM_CONFIG(val) (FIELD_GET(AUXPLL_CONFIG_CFGSTATIC_AUXPLLRANGE_Msk, val) + 3U)
+
+struct clock_control_nrf_auxpll_config {
+	NRF_AUXPLL_Type *auxpll;
+	uint32_t ref_clk_hz;
+	uint32_t ficr_ctune;
+	uint32_t cfg;
```

Comment:
```suggestion
	nrf_auxpll_config_t cfg;
```

Reply:
this requires adding unnecessary runtime overhead to the driver, so this won't be changed. Please, provide an API to configure CFGSTATIC in one go.